recipes/net-snmp/all/conanfile.py

```diff
@@ -0,0 +1,189 @@
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def build_requirements(self):
+        if self._is_msvc:
+            self.build_requires("strawberryperl/5.30.0.1")
+
+    def _patch_msvc(self):
+        ssl_info = self.deps_cpp_info["openssl"]
+        openssl_root = ssl_info.rootpath.replace("\\", "/")
+        search_replace = [
+            (
+                r'$default_openssldir . "\\include"',
+                f"'{openssl_root}/include'"
+            ),
+            (r'$default_openssldir . "\\lib\\VC"', f"'{openssl_root}/lib'"),
+            ("$openssl = false", "$openssl = true")
+        ]
+        if self._is_debug:
+            search_replace.append(("$debug = false", "$debug = true"))
+        if self.options.shared:
+            search_replace.append((
+                "$link_dynamic = false",
+                "$link_dynamic = true"
+            ))
+        if self.options.with_ipv6:
+            search_replace.append(("$b_ipv6 = false", "$b_ipv6 = true"))
+        for search, replace in search_replace:
+            tools.replace_in_file("win32\\build.pl", search, replace)
+        runtime = self.settings.compiler.runtime
+        tools.replace_in_file("win32\\Configure", '"/runtime', f'"/{runtime}')
+        link_lines = "\n".join(
+            f'#    pragma comment(lib, "{lib}.lib")'
+            for lib in ssl_info.libs + ssl_info.system_libs
+        )
+        config = r"win32\net-snmp\net-snmp-config.h.in"
+        tools.replace_in_file(config, "/* Conan: system_libs */", link_lines)
+
+    def _build_msvc(self):
+        if self.should_configure:
+            self._patch_msvc()
+            self.run("perl build.pl", cwd="win32")
+        if self.should_build:
+            with tools.vcvars(self):
+                self.run("nmake /nologo libsnmp", cwd="win32")
+
+    @functools.lru_cache(1)
+    def _configure_autotools(self):
+        disabled_link_type = "static" if self.options.shared else "shared"
+        debug_flag = "enable" if self._is_debug else "disable"
+        ipv6_flag = "enable" if self.options.with_ipv6 else "disable"
+        ssl_path = self.deps_cpp_info["openssl"].rootpath
+        args = [
+            "--with-defaults",
+            "--without-rpm",
+            "--without-pcre",
+            "--disable-agent",
+            "--disable-applications",
+            "--disable-manuals",
+            "--disable-scripts",
+            "--disable-mibs",
+            "--disable-embedded-perl",
+            f"--disable-{disabled_link_type}",
+            f"--{debug_flag}-debugging",
+            f"--{ipv6_flag}-ipv6",
+            f"--with-openssl={ssl_path}",
+        ]
+        autotools = AutoToolsBuildEnvironment(self)
```

Comment:
```suggestion
        autotools = AutoToolsBuildEnvironment(self)
        autotools.libs = []
```

Do not break `configure` if openssl is shared.